test/uk/gov/hmrc/emcstfe/utils/HMRCMarkUtilSpec.scala

```diff
@@ -17,78 +17,215 @@
 package uk.gov.hmrc.emcstfe.utils
 
 import uk.gov.hmrc.emcstfe.fixtures.GetMovementFixture
+import uk.gov.hmrc.emcstfe.models.response.ErrorResponse.MarkCreationError
 import uk.gov.hmrc.emcstfe.support.UnitSpec
 
+import scala.xml.{Elem, XML}
+
 class HMRCMarkUtilSpec extends UnitSpec with GetMovementFixture {
 
   object TestHmrcMark extends HMRCMarkUtil
 
-  "HMRCMarkUtil" should {
+  "HMRCMarkUtil" when {
+    "passed valid XML" should {
+
+      "generate hash from XML" in {
+        // The below XML was taken from going through the submit movement flow in the old EMCS service and capturing the submission XML in Wireshark.
+        // Therefore if the BinarySecurityToken value generated by the old EMCS service is the same as the one our util generates, our util is working correctly
+        // Note that the old EMCS service removes whitespace from its XML before submitting, so we will need to in another util
+        val xml: Elem = XML.loadString(
```

Comment:
You can create an `Elem` without having to load it from a string. You can just have the pure tags now - which will keep it cleaner :) 

You can still do substitution by using `{}` curly braces without a `$`.